runtime/lua/vim/treesitter/dev.lua

```diff
@@ -501,12 +506,50 @@ function M.inspect_tree(opts)
     end,
   })
 
-  api.nvim_create_autocmd('BufHidden', {
+  api.nvim_create_autocmd({ 'WinClosed' }, {
     group = group,
     buffer = buf,
-    once = true,
-    callback = function()
-      close_win(w)
+    desc = 'Update source buffer window if the original was closed',
+    callback = function(ev)
+      -- do nothing if current window is still open
+      if ev.match ~= tostring(win) then
+        return nil
+      end
+
+      local windows = vim.fn.win_findbuf(buf) --[[@as integer[] ]]
+      if #windows == 1 then
+        close_win(w)
+        return true
+      end
+
+      -- Maintain a valid window, if the original window was closed.
+      local _
+      _, win = vim.iter(pairs(windows)):find(function(_, new_win)
+        return new_win ~= win
+      end)
+    end,
+  })
+
+  api.nvim_create_autocmd({ 'WinClosed' }, {
```

Comment:
wait where did this extra handler come from?

this PR is turning into a lot of code, for a pretty minor use-case.

is there a simpler approach we can use? can we enforce a constraint to make things simpler?

Reply:
> wait where did this extra handler come from?

I observed a similar situation that happened if the user opened two InspectTree buffers and closed the original one. Similar to the `BufLeave` events, I think this needs to be duplicated.

>this PR is turning into a lot of code, for a pretty minor use-case.

Yeah, I get it. But it can be really annoying because it forces a user to restart all their window setup. And the repetitive nature of the code is the true reason for so many lines.

>is there a simpler approach we can use? can we enforce a constraint to make things simpler?

The only other way I can think of doing this would be to always validate the windows on the events in which they are used, but that would decrease performance. Do you think this approach could be better?